sys/include/net/ieee802154.h

```diff
@@ -396,6 +396,25 @@ int ieee802154_get_src(const uint8_t *mhr, uint8_t *src, le_uint16_t *src_pan);
  */
 int ieee802154_get_dst(const uint8_t *mhr, uint8_t *dst, le_uint16_t *dst_pan);
 
+/**
+ * @brief  Check whether a frame pass the IEEE 802.15.4 frame filter.
+ *
+ * A frame passes the frame filter only if:
+ * - The PAN ID matches the PAN ID of the frame filter or the broadcast PAN ID
+ * - Either the Short or Extended Address matches the frame filter OR the
+ *   Short Address is the broadcast address.
+ *
+ * @param[in] mhr           MAC header (PSDU)
+ * @param[in] pan           PAN ID of the frame filter.
+ * @param[in] short_addr    Short Address of the frame filter.
+ * @param[in] ext_addr      Extended Address of the frame filter.
+ *
+ * @return 0            if frame passes the frame filter.
+ * @return 1            if frame doesn't pass the frame filter.
+ */
+int ieee802154_dst_filter(const uint8_t *mhr, uint16_t pan,
+                          network_uint16_t short_addr, const eui64_t *ext_addr);
```

Comment:
Just to clarify: `pan` is `uint16_t` but `short_addr` is `network_uint16_t`?

Reply:
yes :(
Unfortunately this is the way how it's presented in many `ieee802154_xxx` functions.

I would actually migrate everything to little endian values. We could the use all `ieee802154_get_xxx` operations without any kind of memcpy / reverse. But we would need to adapt many modules.

Reply:
Lets keep it simple for now, the reversing issue is out of scope of this PR.